dashboard/lib/sample_data.rb

```diff
@@ -0,0 +1,181 @@
+include FactoryGirl::Syntax::Methods
+
+class SampleData
+  TEST_ACCOUNT_PASSWORD = "00secret".freeze
+  @@prng = nil
+
+  # Returns a seeded random number generator for consistent test data
+  def self.prng
+    @@prng ||= Random.new(0)
+    @@prng
+  end
+
+  # Creates sample data
+  def self.seed
+    raise "Should not be run outside of development" unless CDO.rack_env?(:development)
+
+    # Create a test teacher
+    teacher = create_teacher "testteacher@code.org", "TestTeacher Codeberg"
+
+    # Create normal-sized and large sections for each of CSF, CSD and CSP
+
+    create_section({teacher: teacher, name: "CSF 1",
+      login_type: Section::LOGIN_TYPE_PICTURE, grade: 2, age_min: 7,
+      age_max_inclusive: 9, script_name: "coursea-2018", num_students: 30,
+      use_imperfect_results: true}
+      )
+
+    create_section({teacher: teacher, name: "CSF 2 (Large)",
+      login_type: Section::LOGIN_TYPE_PICTURE, grade: 2, age_min: 7,
+      age_max_inclusive: 9, script_name: "coursea-2018", num_students: 150,
+      use_imperfect_results: true}
+      )
+
+    create_section({teacher: teacher, name: "CSD 1",
+      login_type: Section::LOGIN_TYPE_EMAIL, grade: 7, age_min: 13,
+      age_max_inclusive: 14, script_name: "csd1-2018", num_students: 30,
+      use_imperfect_results: false}
+      )
+
+    create_section ({teacher: teacher, name: "CSD 2 (Large)",
+      login_type: Section::LOGIN_TYPE_EMAIL, grade: 7, age_min: 13,
+      age_max_inclusive: 14, script_name: "csd1-2018", num_students: 150,
+      use_imperfect_results: false}
+      )
+
+    create_section ({teacher: teacher, name: "CSP 1",
+      login_type: Section::LOGIN_TYPE_EMAIL, grade: 10, age_min: 14,
+      age_max_inclusive: 16, script_name: "csp1-2018", num_students: 30,
+      use_imperfect_results: false}
+      )
+
+    create_section ({teacher: teacher, name: "CSP 2 (Large)",
+      login_type: Section::LOGIN_TYPE_EMAIL, grade: 10, age_min: 14,
+      age_max_inclusive: 16, script_name: "csp1-2018", num_students: 150,
+      use_imperfect_results: false}
+      )
+  end
+
+  # will delete the teacher and all of the teacher's sections and students
+  # and recreate.
+  def self.create_teacher(email, name)
+    # Delete any existing test data
+    user = User.find_by_email_or_hashed_email(email)
+    unless user.nil?
+      user.sections.each do |section|
+        # Delete all students in each section
+        section.followers.each do |follower|
+          student_user = User.find_by_id(follower.student_user_id)
+          raise "Not a test user" unless student_user.name.include? "Codeberg"
```

Comment:
This seems a dicey. We may override the test user's name, and what's to keep a real user from being named Codeberg?

I'd prefer a more reliable way of differentiating these test users if we want to keep them apart. 
What about adding a serialized field to `User`, only present when true, called something like `test_user` that we always set through the factory (can be default factory or a trait). 

As a serialized property it will be stored in the existing `properties` JSON column and not require a data migration.

Reply:
I'm _mostly_ okay with this given the other defenses in place to only run this in development, and I'm not thrilled with the idea of adding a properties column for this purpose (it may not require a migration but it's still another test-specific support in production code).  That said, I think the following would be an improvement:

- Whatever identifying string we use for a test user should be in a constant in this file to minimize the chance of typos.
- Setting a custom username may be more reliable, since display name can be changed via a normal user workflow and one can imagine changing the display name of one of these users in a test scenario and subsequently failing to delete the user.

Reply:
If we're sure this will stay only in development, then it's ok but then I agree use a constant in this file as the name (or name prefix) and make it something less like a real name and different from the FactoryBot default of "User n Codeberg".

If we think this could ever run on production, then it scares me to automatically hard delete users and I do want more protection even if it means having "test-specific support" in production. We could call it `seeded_user` so it's less test specific. 

I'd like to see a unit test for this too - make sure we delete the users we expect and only them.